app/controllers/proposals_controller.rb

```diff
@@ -74,6 +75,17 @@ def update
     end
   end
 
+  def toogle_favorite
+    user = User.find(params[:favourite_user_id])
+    users = @event.favourite_users
+    if users.include? user
+      @event.favourite_users.delete(user)
+    else
+      @event.favourite_users << User.find(params[:favourite_user_id])
+    end
+    render json: {}
```

Comment:
what could go wrong?

`delete` and `<<` can not fail

we could check that the record was updated or deleted and then render an error.